crates/orchestrator/src/jobs/proving_job/mod.rs

```diff
@@ -54,15 +50,15 @@ impl Job for ProvingJob {
     }
 
     async fn process_job(&self, config: &Config, job: &mut JobItem) -> Result<String, JobError> {
-        // TODO: allow to download PIE from storage
-        let cairo_pie_path = job
-            .metadata
-            .get(JOB_METADATA_CAIRO_PIE_PATH_KEY)
-            .map(|s| PathBuf::from_str(s))
-            .ok_or_else(|| ProvingError::CairoPIEWrongPath { internal_id: job.internal_id.clone() })?
-            .map_err(|_| ProvingError::CairoPIENotReadable)?;
-
-        let cairo_pie = CairoPie::read_zip_file(&cairo_pie_path).map_err(|_| ProvingError::CairoPIENotReadable)?;
+        // Cairo Pie path in s3 storage client
+        let cairo_pie_path = job.internal_id.to_string() + "/pie.zip";
+        let cairo_pie_file = config
+            .storage()
+            .get_data(&cairo_pie_path)
+            .await
+            .map_err(|e| ProvingError::CairoPIEFileFetchFailed(e.to_string()))?;
+        let cairo_pie =
+            CairoPie::from_bytes(cairo_pie_file.to_vec().as_slice()).map_err(|_| ProvingError::CairoPIENotReadable)?;
```

Comment:
still not reading `e` over here? what's the confusion?

Reply:
we talked about this right ? sirf upar wale me e read krna tha. neeche waale me you told that we need to pass CairoPieNoReadable

Reply:
No. It was that we can do `Other` error  (we're using `CairoPIEFileFetchFailed` now but that's fine) on the first one and `CairoPieNotReadable` in the lower. As a thumb rule, we should always pass the error objects because if this breaks tomorrow, we won't have context of the actual error.